### PR TITLE
Refactor Rego policies

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Refactor Rego policies

## why
* Move policies to `catalog`
* Make code more generic, stop adding the policy feature flags for each individual policy
* Now we can add any number of policies to the `catalog`, and because of the naming convention (format of the filename is `$type.$friendly_name.rego`), they will be created with the correct (automatically detected) type and name and attached to the Spacelift stacks

## references
* https://github.com/cloudposse/terraform-spacelift-cloud-infrastructure-automation/pull/41#issuecomment-862095925
* https://github.com/cloudposse/terraform-spacelift-cloud-infrastructure-automation/pull/6#discussion_r560685455

### CODE REVIEW

**Review by jamengual — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -152,32 +152,36 @@ variable "administrative_trigger_policy_enabled" {
...
+  default = [
+    "access.default",
+    "git_push.default",
+    "plan.default",
+    "trigger.dependencies",
+    "trigger.retries"
+  ]
+}
+
+variable "policies_attach" {
```
Comment (osterman) [inline variables.tf:167]:
```suggestion
variable "policies_enabled" {
```

File: variables.tf [hunk 2]
```diff
@@ -152,32 +152,36 @@ variable "administrative_trigger_policy_enabled" {
...
-  type        = bool
-  description = "Flag to enable/disable the stack access policy"
-  default     = true
-}
-
-variable "push_policy_enabled" {
-  type        = bool
-  description = "Flag to enable/disable the stack push policy"
-  default     = true
+variable "policies_create" {
```
Comment (osterman) [inline variables.tf:155]:
```suggestion
variable "policies_available" {
```

File: variables.tf [hunk 3]
```diff
@@ -152,32 +152,36 @@ variable "administrative_trigger_policy_enabled" {
...
+  type        = string
+  description = "Path to the catalog of default policies"
+  default     = "catalog/policies"
 }
 
-variable "plan_policy_enabled" {
-  type        = bool
-  description = "Flag to enable/disable the stack plan policy"
-  default     = true
+variable "additional_policy_ids" {
```
Comment (osterman) [inline variables.tf:183]:
```suggestion
variable "additional_policies_enabled" {
```

Comment (aknysh) [reply variables.tf:183]: I think `additional_policy_ids` better convey the meaning: the caller creates additional policies in their component and provides their IDs to attach to each Spacelift stack (they are enabled by definition)

**Review by osterman — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Pull request merged**